### DISCUSSION TITLE
Setting recording path or filename for Source Record

### DISCUSSION THREAD

**Discussion by therentabrain** [WarmUpTill/SceneSwitcher#1073] [Q&A] [answered]
The Source Record filter is notoriously simultaneously wonderful and frustrating.  I want to control the filename it's going to record to, or failing that, at least control the folder name.  SceneSwitcher doesn't seem to have access to that field.  In fact, the values it does seem to have access to are kind of nonsensical and have no value.  I imagine the plugin doesn't use properties in a standards compliant way.

I thought once upon a time, there was a way to directly edit the JSON settings of things using SceneSwitcher.  Then at least I could replace the filename in the JSON and impose that on the filter fresh, every session.  Is that not possible?  Any other solutions?

Thanks!

**Comment by therentabrain** [comment]
Here's the json for it:

```
{
          "versioned_id": "source_record_filter",
          "muted": false,
          "balance": 0.5,
          "deinterlace_field_order": 0,
          "push-to-talk-delay": 0,
          "hotkeys": {},
          "push-to-mute-delay": 0,
          "volume": 1,
          "flags": 0,
          "private_settings": {},
          "id": "source_record_filter",
          "push-to-talk": false,
          "push-to-mute": false,
          "monitoring_type": 0,
          "uuid": "0e6b11e2-7e92-4640-b0dc-0ccead9efcee",
          "sync": 0,
          "name": "Source Record",
          "mixers": 0,
          "prev_ver": 503382017,
          "enabled": false,
          "settings": {
            "lookahead": false,
            "gpu": 0,
            "bitrate": 5000,
            "psycho_aq": false,
            "preset": "veryfast",
            "encoder_type": "nvh264enc",
            "record_mode": 4,
            "filename_formatting": "%CCYY %MM %DD %hh-%mm-%ss_mysecondarycam",
            "keyint_sec": 10,
            "different_audio": true,
            "encoder": "x264",
            "tune": "",
            "path": "C:/recordings",
            "multipass": "disabled",
            "audio_track": 1,
            "rate_control": "CBR",
            "preset2": "p4"
          },
          "deinterlace_mode": 0
        }
```

**Comment by WarmUpTill** [comment, marked as answer]
You should be able to modify just the recording path setting using the "Filter" action's `Set settings` option with this build:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/8425094585?pr=1072

![grafik](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/9e4c40f2-b167-44a4-8cc2-2c4160f9bab8)

You can also still modify the raw json, if you prefer to do it that way, using the `Set settings JSON string` option:
![grafik](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/13fbfb78-d693-43fc-869f-4c35140f313e)

Let me know if you run into any issues or if you have further questions! :)

**Reply by therentabrain** [reply]
Wondrous!

**Comment by therentabrain** [comment]
Awesome - but it says cancelled, should I wait?

**Reply by WarmUpTill** [reply]
No need to wait.
Only the MacOS build was canceled.
If you need the MacOS version as well, it can be found here:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/8425968604

**Reply by therentabrain** [reply]
@WarmUpTill I asked in my other thread about getting the 2 digit hours, minutes, day, month etc.  Is there a way to do that?  For file naming column consistency.

**Reply by WarmUpTill** [reply]
That is possible, but a bit complicated.
To make it easier I added the option to pad the values of a variable:
![grafik](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/9cb8fb17-a114-469c-8896-e682bad9795a)

A build will be available here in a few minutes:
https://github.com/WarmUpTill/SceneSwitcher/actions/runs/8452471529?pr=1072

Let me know if that works as you expect it to :)

**Reply by therentabrain** [reply]
I finally got back to this!  It works great.  We use the timestamp of when OBS starts to set the recording filenames (so the various sources file together nicely) and also use it to set the clock - which didn't require the padding, but we do it in the same macro now.

For anyone who wants to mess with it, here it is!

[macro.txt](https://github.com/WarmUpTill/SceneSwitcher/files/15376682/macro.txt)

And thank you for this ever increasingly wonderful tool :)
